src/Hosting/Hosting/src/Internal/HostingApplication.cs

```diff
@@ -122,13 +122,26 @@ internal class Context
         {
             public HttpContext? HttpContext { get; set; }
             public IDisposable? Scope { get; set; }
-            public Activity? Activity { get; set; }
+            public Activity? Activity
+            {
+                get => HttpActivityFeature?.Activity;
+                set
+                {
+                    if (HttpActivityFeature is null)
+                    {
+                        HttpActivityFeature = new ActivityFeature(value!);
```

Comment:
HostingApplication's aren't pooled by default, right? So this ends up being an allocation per request?

Reply:
Kestrel is pooled I think (`IHostContextContainer`), but the other servers don't look like they implement the feature.

Reply:
Why do we need any of this code? Is this for IIS and HTTP.sys?

Reply:
The feature is cached on this Context to be reused across connections (in Kestrel at least)

Reply:
That part I know, I'm trying to figure out where this happens.

Reply:
Do you just want the feature to be newed up in the ctor of Context? The reason I avoided that is because if there isn't a listener (no Activity) the feature doesn't need to be allocated.

Reply:
I'm fine with this I just wanted to confirm that the case where it allocates per request is in IIS and Http.sys where we don't default this feature.

Reply:
Correct